File: api/news.py
"""
财经新闻 API - Vercel Serverless Function
"""
from flask import Flask, jsonify
import requests
import json

app = Flask(__name__)

@app.route('/api/news', methods=['GET'])
def get_news():
    """获取财经新闻（东方财富快讯）"""
    try:
        # 使用东方财富财经快讯 API
        resp = requests.get(
            'https://newsapi.eastmoney.com/kuaixun/v1/getlist_102_ajaxResult_50_1_.html',
            timeout=10
        )
        
        # 解析 JSONP 格式：var ajaxResult={...}
        text = resp.text
        if text.startswith('var ajaxResult='):
            json_str = text.replace('var ajaxResult=', '')
            data = json.loads(json_str)
            
            if data.get('LivesList'):
                news_list = []
                for item in data['LivesList'][:20]:
                    news_list.append({
                        'title': item.get('title', ''),
                        'digest': item.get('digest', ''),
                        'url': item.get('url_w', ''),
                        'source': '东方财富',
                        'publish_time': item.get('showtime', '')
                    })
                return jsonify({
                    'success': True,
                    'source': 'eastmoney',
                    'items': news_list
                })
        
        return jsonify({'success': False, 'error': '解析新闻数据失败'})
    except Exception as e:
        return jsonify({'success': False, 'error': str(e)})

# Vercel 需要这个 handler
handler = app

File: api/news.js
export default async function handler(req, res) {
  res.setHeader('Access-Control-Allow-Origin', '*');
  
  if (req.method !== 'GET') {
    return res.status(405).json({ success: false, error: 'Method not allowed' });
  }
  
  try {
    const response = await fetch(
      'https://newsapi.eastmoney.com/kuaixun/v1/getlist_102_ajaxResult_50_1_.html'
    );
    
    const text = await response.text();
    
    if (text.startsWith('var ajaxResult=')) {
      const jsonStr = text.replace('var ajaxResult=', '');
      const data = JSON.parse(jsonStr);
      
      if (data.LivesList) {
        const newsList = data.LivesList.slice(0, 20).map(item => ({
          title: item.title || '',
          digest: item.digest || '',
          url: item.url_w || '',
          source: '东方财富',
          publish_time: item.showtime || ''
        }));
        
        return res.status(200).json({
          success: true,
          source: 'eastmoney',
          items: newsList
        });
      }
    }
    
    return res.status(500).json({ success: false, error: '解析新闻数据失败' });
    
  } catch (error) {
    return res.status(500).json({ success: false, error: error.message });
  }
}
